api/tasks.js

```diff
@@ -8,14 +8,53 @@ const Sentry = require("@sentry/node");
 /* This router supports superuser routes, such as updating the status of users */
 
 router.use(utils.authMiddleware);
-router.use(utils.requireAdmin);
 
-router.get("/list", async (req, res) => {
-  const tasks = await models.Task.findAll();
-  return res.json({ success: tasks });
+router.get("/list", (req, res, next) => {
+    // fallback to check admin if role doesn't exist in req
+    if (!req.user.role || req.user.role === "admin") {
+      utils.requireAdmin(req, res, next)
+    } else if (req.user.role === "volunteer") {
+      utils.requireVolunteer(req, res, next)
+    } else {
+      utils.requireSponsor(req, res, next)
+    }
```

Comment:
I'm not sure if you can use express middleware like this, wanna double check this is a pattern that works?

Reply:
Yeah I double checked it: https://stackoverflow.com/questions/21271492/conditionally-use-a-middleware-depending-on-request-parameter-express
It makes sense since next is just a callback function, and if it doesn't get called, it'll reject